subsys/bluetooth/mesh/scene_srv.c

```diff
@@ -0,0 +1,716 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <bluetooth/mesh/models.h>
+#include <sys/byteorder.h>
+#include "model_utils.h"
+#include "mesh/net.h"
+#include "mesh/access.h"
+
+#define BT_DBG_ENABLED IS_ENABLED(CONFIG_BT_MESH_DEBUG_MODEL)
+#define LOG_MODULE_NAME bt_mesh_scene_srv
+#include "common/log.h"
+
+#define SCENE_PAGE_SIZE SETTINGS_MAX_VAL_LEN
+
+struct __packed scene_data {
+	uint8_t len;
+	uint8_t elem_idx;
+	uint16_t id;
+	uint8_t data[];
+};
+
+static char *scene_path(char buf[9], uint16_t scene, bool vnd, uint8_t page)
```

Comment:
`[9]` doesn't make any sense as it will anyway be converted into a pointer. I'm recalling a letter from one guy...: https://lkml.org/lkml/2015/9/3/428

Reply:
True, and I enjoyed Linus' email :smile: I think he would be quite upset if he saw the Zephyr codebase though, there is a lot of this going around as a form of parameter documentation. IMO, anyone using `sizeof(parameter)` on a pointer are in deep trouble regardless of whether the pointer has an array declaration or not. I can remove it though, it doesn't make a big difference.